core/src/main/java/fr/sncf/osrd/infra/implementation/tracks/undirected/UndirectedInfraBuilder.java

```diff
@@ -116,6 +117,10 @@ private void addRemainingLinks(RJSInfra infra) {
                         oldDstNode
                 ));
             }
+            if (link.id == null) {
+                // Forcing a unique ID avoids node equality troubles, and makes debugging easier
+                link.id = String.format("generated_%d", generatedID++);
```

Comment:
do you think issuing a warning would make sense?

Reply:
I don't think unlabeled links are much of a concern